### PR TITLE
V2 restructure

### PR DESCRIPTION
**Pull request by lf-jason-jia**
1. Normalize CI naming
2. Update CI branch matching
3. Update tag name to include API version (to avoid confusion)
4. Update project description to include branching information
5. Update project name (suffix v2, that's common in Maven)
6. Reset version number to 1.0.0 since we are publishing a new project
7. Update documentation publishing CI
8. Update documentation link

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,13 +2,16 @@ name: lf-repository-api-client-java-CI
...
   pull_request:
-    branches: [ '\d+.x' ]
+    branches:
+      - v2
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 env:
+  API_VERSION: 'v2'
   VERSION_PREFIX: '2.0.3'
```
Comment (jason-chan-lf) [inline .github/workflows/main.yml]: can you also reset the version here

File: pom.xml [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
   <groupId>com.laserfiche</groupId>
-  <artifactId>lf-repository-api-client</artifactId>
+  <artifactId>lf-repository-api-client-v2</artifactId>
   <packaging>jar</packaging>
-  <name>Laserfiche Repository API Client</name>
-  <version>2.0.3</version>
+  <name>Laserfiche Repository API Client V2</name>
+  <version>1.0.0</version>
   <url>https://github.com/Laserfiche/lf-repository-api-client-java</url>
   <description>Java implementation of various foundational APIs for Laserfiche, including Repository APIs such as
```
Comment (jason-chan-lf) [inline pom.xml]: can you update the description to make it clear this package access the v2 laserfiche repository apis

**Review by samadpaydar3 — DISMISSED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (lf-jason-jia) [reply .github/workflows/main.yml]: Fixed

File: pom.xml (hunk 2 above)
Comment (lf-jason-jia) [reply pom.xml]: Fixed

**Review by jason-chan-lf — APPROVED** [review]

**Pull request merged**